client/verta/verta/endpoint/_endpoint.py

```
@@ -508,6 +508,9 @@ def _set_kafka_settings(self, stage_id, kafka_settings):
                 "`kafka_settings` must be of type verta.endpoint.KafkaSettings,"
                 " not {}".format(type(kafka_settings))
             )
+        if kafka_settings._cluster_config_id is None:
+            config = kafka.list_kafka_configurations(self._conn)[0]
+            kafka_settings._cluster_config_id = config.get("id", None) if config else None
```

What error does the backend return if we pass `None` as the config ID?

---

I'm asking because I feel like if `config` doesn't contain `"id"` (`config.get("id", None)`), or if `config` is somehow empty (`if config`), tyhen that should be an error condition rather than just defaulting to `None`.

But if the backend will return an error a couple lines below, then that's probably fine.

---

`list_kafka_configurations` itself will raise an HTTPError if there is no valid config, but I don't know what happens if we actually manage to pass `None` as the ID to back-end.  I'll test that out real quick

---

It looks like passing `None` along to the backend throws
`requests.exceptions.HTTPError: 500 Server Error: Kafka config ID does not correlate with valid kafka configuration for url: https://ewagner.dev.verta.ai/api/v1/deployment/stages/237/kafka` in the next step.

A 500 seems appropriate, since this would have to be caused by us breaking the "contract" in the API response.

---

How do you feel about this:

I think it's a better practice for the client to explicitly raise an error here saying "We expected a non-empty config with the backend" or "The config should've contained an ID, but didn't". Rather than silently set it to `None` and rely on a subsequent call to the backend to raise a `500`.

We already know our program's state here is going to result in an issue; we shouldn't have it continue running.

And I don't think a client should rely on the backend to raise an error for an issue that itself originated from the backend.

---

Ok, cool.  No objections to that and it gives us an opportunity to be a bit more specific with the error message.  I'll ping you when that update is in place shortly.

---

Changes and related tests are now included